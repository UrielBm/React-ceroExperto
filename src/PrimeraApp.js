import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
const PrimeraApp=({saludo, subtitulo}) => {
    // const saludo2="Hola banda";
    // const saludo={
    //     nombre: 'Fernando',
    //     edad:34,
    //     entidad: 'Hidalgo',
    // };

    return (
        <Fragment>
        <h1>{saludo}</h1>
        {/* <pre>{JSON.stringify(saludo,null, 3 )}</pre> */}
        <p>{subtitulo}</p>
        </Fragment>
    ); 
}
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
PrimeraApp.defaultProps={
    subtitulo: 'Prueba tu primera App'
}
export default PrimeraApp;