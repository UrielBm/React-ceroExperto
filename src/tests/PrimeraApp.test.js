import React from 'react';
import '@testing-library/jest-dom';
// import {render} from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import { shallow } from 'enzyme';
describe('Evaluendo componente de reacr PrimeraApp.js', () => {
    // Test con // import '@testing-library/jest-dom/extend-expect'; de setupTest.js 
    // test('Debe mostrar el mensaje: Hola,qué onda la banda? ', () => {
    //     const mensaje= 'Hola,qué onda la banda?';
    //     const {getByText} = render(<PrimeraApp saludo={mensaje} />);
    //     expect(getByText(mensaje)).toBeInTheDocument();
    // });
    test('debe de mostrare <PrimeraApp /> correctamente ', () => {
        const saludo='Hola,qué onda la banda?';
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    });
    test('Envio de subtitulos enviado por props en <PrimeraApp /> ', () => {
        const subtitulos= 'Todo esta bien';
        const saludo='Hola,qué onda la banda?';
        const wrapper = shallow(<PrimeraApp saludo={saludo} subtitulo={subtitulos} />);
        const textoParrafo= wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulos);
    })
    
    
})
