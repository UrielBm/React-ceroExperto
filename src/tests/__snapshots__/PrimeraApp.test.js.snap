// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Evaluendo componente de reacr PrimeraApp.js debe de mostrare <PrimeraApp /> correctamente  1`] = `
<Fragment>
  <h1>
    Hola,qu√© onda la banda?
    !!!
  </h1>
  <p>
    Prueba tu primera App
  </p>
</Fragment>
`;
